src/Bicep.Local.Deploy.IntegrationTests/EndToEndDeploymentTests.cs

```diff
@@ -138,14 +143,24 @@ extension http
   }
 }
 """;
-                return Task.FromResult(new LocalExtensibilityOperationResponse(new Resource(req.Type, req.ApiVersion, identifiers, req.Properties, "Succeeded"), null));
+                return Task.FromResult(new Extension.Protocol.LocalExtensionOperationResponse(new Resource(req.Type, req.ApiVersion, identifiers, req.Properties, "Succeeded"), null));
             });
 
-        var dispatcher = BicepTestConstants.CreateModuleDispatcher(services.Build().Construct<IServiceProvider>());
-        await using LocalExtensibilityHostManager extensibilityHandler = new(dispatcher, uri => Task.FromResult(providerMock.Object));
-        await extensibilityHandler.InitializeExtensions(result.Compilation);
+        var localExtensionFactory = StrictMock.Of<ILocalExtensionFactory>();
+        localExtensionFactory
+            .Setup(x => x.CreateLocalExtensionAsync(It.IsAny<LocalExtensionKey>(), It.IsAny<Uri>()))
+            .ReturnsAsync(providerMock.Object);
 
-        var localDeployResult = await LocalDeployment.Deploy(extensibilityHandler, templateFile, parametersFile, TestContext.CancellationTokenSource.Token);
+        var moduleDispatcher = BicepTestConstants.CreateModuleDispatcher(services.Build().Construct<IServiceProvider>());
+
+        var extensionFactoryManager = new LocalExtensionFactoryManager(localExtensionFactory.Object);
```

Comment:
Most of the code remained the same, one of the biggest differences is the fact that there is an interface to create extension instances - `ILocalExtensionFactory` which is meant for clients to implement it - see `LocalDeployCommand.cs` and the other difference if `LocalExtensionFactoryManager` this class is in charge of initializing local extensions and load them on-demand and only once.